.clang-format

```diff
@@ -17,4 +17,7 @@ AllowShortLoopsOnASingleLine: false
 BinPackParameters: false
 BreakBeforeBinaryOperators: true
 BreakBeforeTernaryOperators: true
-ContinuationIndentWidth: 4
+ContinuationIndentWidth: 2
+AlignEscapedNewlines: DontAlign
+SortIncludes: false
```

Comment:
Curious why `SortIncludes: false`? I personally find this useful, especially in files with lots of of `#includes`, to see if a given header is already present.

Reply:
Because of the clint warning

 > Found C system header after other header. Should be: c system, c++ system, other. [build/include_order] [4]

If we add a `SortInlcudes: false` then we only need to sort them once manually and then be done with it, which seemed like the simplest solution to me. It's possible to sort includes by assigning priority with regexes but that looked like big trouble.

Reply:
+1, sorting also helps avoid merge conflicts

Reply:
> Because of the clint warning
> 
> > Found C system header after other header. Should be: c system, c++ system, other. [build/include_order] [4]
> 
> If we add a `SortInlcudes: false` then we only need to sort them once manually and then be done with it, which seemed like the simplest solution to me. It's possible to sort includes by assigning priority with regexes but that looked like big trouble.

If you separate the system headers from the project headers with a blank line, does clang-format still auto sort all of them into one big list, or does it respect the separation?

Reply:
> It's possible to sort includes by assigning priority with regexes but that looked like big trouble.

If we can sort the `#include <...>` includes first, that should be good enough. And then we can remove/relax the clint rule if it's still not happy.

Reply:
> > Because of the clint warning
> > > Found C system header after other header. Should be: c system, c++ system, other. [build/include_order] [4]
> > 
> > 
> > If we add a `SortInlcudes: false` then we only need to sort them once manually and then be done with it, which seemed like the simplest solution to me. It's possible to sort includes by assigning priority with regexes but that looked like big trouble.
> 
> If you separate the system headers from the project headers with a blank line, does clang-format still auto sort all of them into one big list, or does it respect the separation?

Unfortunately not.

Reply:
> > It's possible to sort includes by assigning priority with regexes but that looked like big trouble.
> 
> If we can sort the `#include <...>` includes first, that should be good enough. And then we can remove/relax the clint rule if it's still not happy.

I think that's possible. Two possible solutions:

1. Add `// clang-format off/on` before and after the `#include <...>` block.
2. Prioritize `#include <...>` over other includes with the help of regex rules.

I'd probably go for the option 2 _if_ that's the only order that's important. If we start ordering three or more categories then probably option 1 (maybe it's my regex-phobia talking, so may not be entirely unbiased :) )

Reply:
> I'd probably go for the option 2 _if_ that's the only order that's important. If we start ordering three or more categories then probably option 1 (maybe it's my regex-phobia talking, so may not be entirely unbiased :) )

That should be it since we're (mercifully) not using C++, so there's no C/C++ header distinction, only system/user.

Reply:
@gpanders Aaaaand it turns out I was wrong, there's absolutely an option for sorting includes according to blocks. Should've investigated more before answering :). Anyway, still went with the regex option since it'll automatically create the correct include blocks regardless of how they were previously ordered.